docs/root/configuration/http/http_filters/composite_filter.rst

```diff
@@ -0,0 +1,45 @@
+.. _config_http_filters_composite:
+
+Composite Filter
+================
+
+The composite filter allows delegating filter actions to a filter specified by a
+:ref:`match result <arch_overview_matching_api>`. The purpose of this is to allow different filters
+or filter configurations to be selected based on the incoming request, allowing for more dynamic
+configuration that could become prohibitive when making use of per route configurations (e.g.
+because the cardinality would cause a route table explosion).
+
+The filter does not do any kind of buffering, and as a result it must be able to instantiate the
+filter it will delegate to before it receives any callbacks that it needs to delegate. Because of
+this, in order to delegate all the data to the specified filter, the decision must be made based
+on just the request headers.
+
+Delegation can fail for a few reasons: the match result was not ready in time for the filter
```

Comment:
This can't happen anymore given that we require only request headers, right?

Reply:
Yep, will rephrase